pkg/csi/service/wcpguest/controller.go

```diff
@@ -736,6 +740,10 @@ func controllerUnpublishForBlockVolume(ctx context.Context, req *csi.ControllerU
 			break
 		}
 		if err := c.vmOperatorClient.Get(ctx, vmKey, virtualMachine); err != nil {
+			if errors.IsNotFound(err) {
```

Comment:
Comment on line#795. While watching for VM events, if we see delete event for the VM, should we not return success there as well?

Reply:
@SandeepPissay yes thats true, addressed in latest commit